src/buildHTML.js

```diff
@@ -450,20 +436,16 @@ groupTypes.genfrac = function(group, options) {
     const dstyle = style.fracDen();
     let newOptions;
 
-    newOptions = options.withStyle(nstyle);
-    const numer = buildGroup(group.value.numer, newOptions);
-    const numerreset = makeSpan([style.reset(), nstyle.cls()],
-        [numer], newOptions);
+    newOptions = options.havingStyle(nstyle);
+    const numerm = buildGroup(group.value.numer, newOptions, options);
 
-    newOptions = options.withStyle(dstyle);
-    const denom = buildGroup(group.value.denom, newOptions);
-    const denomreset = makeSpan([style.reset(), dstyle.cls()],
-        [denom], newOptions);
+    newOptions = options.havingStyle(dstyle);
+    const denomm = buildGroup(group.value.denom, newOptions, options);
 
     let ruleWidth;
     if (group.value.hasBarLine) {
         ruleWidth = fontMetrics.metrics.defaultRuleThickness /
-            options.style.sizeMultiplier;
+            options.sizeMultiplier;
```

Comment:
It might be better to call `makeLineSpan` here and then use its `height` below as opposed to calculating the `ruleWidth` here and inside of `makeLineSpan`.